src/Stratis.Bitcoin.Features.Consensus/CoinViews/Coindb/Coindb.cs

```diff
@@ -129,46 +149,18 @@ public void Initialize(ChainedHeader chainTip)
                     }
                 }
             }
-
-            EnsureCoinDatabaseIntegrity(chainTip);
-
-            Block genesis = this.network.GetGenesis();
-
-            if (this.GetTipHash() == null)
-            {
-                using (var batch = this.coinDb.GetWriteBatch())
-                {
-                    this.SetBlockHash(batch, new HashHeightPair(genesis.GetHash(), 0));
-                    batch.Write();
-                }
-            }
-
-            this.logger.LogInformation("Coinview initialized with tip '{0}'.", this.persistedCoinviewTip);
         }
 
-        private void EnsureCoinDatabaseIntegrity(ChainedHeader chainTip)
+        /// <summary>Just check the integrity. Coin view performs the sync with the chain tip.</summary>
+        private void EnsureCoinDatabaseIntegrity()
         {
             this.logger.LogInformation("Checking coin database integrity...");
 
-            var heightToCheck = chainTip.Height;
-
-            // Find the height up to where rewind data is stored above chain tip.
-            do
-            {
-                heightToCheck += 1;
-
-                byte[] row = this.coinDb.Get(rewindTable, BitConverter.GetBytes(heightToCheck).Reverse().ToArray());
-                if (row == null)
-                    break;
-
-            } while (true);
-
-            for (int height = heightToCheck - 1; height > chainTip.Height;)
+            if (this.GetTipHash() == null)
             {
-                this.logger.LogInformation($"Fixing coin database, deleting rewind data at height {height} above tip '{chainTip}'.");
-
-                // Do a batch of rewinding.
-                height = RewindInternal(height, new HashHeightPair(chainTip)).Height;
```

Comment:
Is this the redundant code mentioned in the title of the PR? If so, where else is the rewinding on startup being done?

Or are we just changing the approach and rebuilding the entire database if the node was not shut down properly?

Reply:
Yes, the rewinding already happens in the  `CachedCoinView`'s `Sync` method.